### PR TITLE
Fire and smoke example

### PR DESCRIPTION
**Pull request by mikepiecuch**
This is my first contribution to three.js.  I think this library is amazing and I want to give back.

This is an example of fire and smoke based on the research paper:

"Real-Time Fluid Dynamics for Games" by Jos Stam
http://www.dgp.toronto.edu/people/stam/reality/Research/pdf/GDC03.pdf

### CODE REVIEW

**Comment by looeee** [comment]
For testing: https://raw.githack.com/mikepiecuch/three.js/fire_example/examples/webgl_fire.html

File: examples/webgl_fire.html [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+				scene.background = new THREE.Color( 0x000000 );
+
+				var ambientLight = new THREE.AmbientLight( 0xcccccc, 0.4 );
+				scene.add( ambientLight );
+
+				var pointLight = new THREE.PointLight( 0xffffff, 0.8 );
+				camera.add( pointLight );
+				scene.add( camera );
+
+				var plane = new THREE.PlaneGeometry( 20, 20 );
```
Comment (Mugen87) [inline examples/webgl_fire.html]: Please use `BufferGeometry` at all places. So in this case `PlaneBufferGeometry`.

File: examples/webgl_fire.html [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+					debug: false
+				} );
+				fire.position.z = - 2;
+				scene.add( fire );
+
+				renderer = new THREE.WebGLRenderer( { antialias: true, alpha: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.autoClear = false;
+				renderer.antialias = true;
```
Comment (Mugen87) [inline examples/webgl_fire.html]: This line seems redundant to me.

File: examples/js/Fire.js [hunk 3]
```diff
@@ -0,0 +1,1072 @@
...
+	}
+
+
+	this.fieldScene = new THREE.Scene();
+	this.fieldScene.background = new THREE.Color( 0x000000 );
+
+	this.orthoCamera = new THREE.OrthographicCamera( textureWidth / - 2, textureWidth / 2, textureHeight / 2, textureHeight / - 2, 1, 2 );
+	this.orthoCamera.position.z = 1;
+
+	this.fieldGeometry = new THREE.PlaneGeometry( textureWidth, textureHeight );
```
Comment (Mugen87) [inline examples/js/Fire.js]: `PlaneBufferGeometry`.

File: examples/js/Fire.js [hunk 4]
```diff
@@ -0,0 +1,1072 @@
...
+		stencilBuffer: false
+	};
+
+
+	this.field0 = new THREE.WebGLRenderTarget( textureWidth, textureHeight, parameters );
+
+	this.field0.background = new THREE.Color( 0x000000 );
+
+	this.field1 = new THREE.WebGLRenderTarget( textureWidth,
+											   textureHeight,
```
Comment (Mugen87) [inline examples/js/Fire.js]: Some formatting issues.

File: examples/js/Fire.js [hunk 5]
```diff
@@ -0,0 +1,1072 @@
...
+	this.setSourceMap = function ( texture ) {
+
+		this.sourceMaterial.uniforms.sourceMap.value = texture;
+
+	};
+
+	var parameters = {
+		minFilter: THREE.NearestFilter,
+		magFilter: THREE.NearestFilter,
+		format: THREE.RGBAFormat,
```
Comment (Mugen87) [inline examples/js/Fire.js]: The default is already `THREE.RGBAFormat`.

**Review by Mugen87 — CHANGES_REQUESTED** [review]

**Comment by Mugen87** [comment]
A better place for `Fire.js` might be `examples/js/objects/`.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**